flang/lib/Lower/FIRBuilder.cpp

```diff
@@ -166,21 +166,40 @@ struct CharacterOpsBuilderImpl {
       return fir::ReferenceType::get(getCharacterType());
     }
 
-    bool isConstant() const { return data.getType().isa<fir::SequenceType>(); }
+    bool isNotInMemory() const {
+      return data.getType().isa<fir::SequenceType>() ||
+             data.getType().isa<fir::CharacterType>();
+    }
+
+    std::optional<fir::SequenceType::Extent> getCompileTimeLenght() const {
+      auto type = data.getType();
+      if (type.isa<fir::CharacterType>())
+        return 1;
```

Comment:
I thought I wrote some "get size in bytes" sort of function someplace a long time ago...

Anyway, have to look at the KIND of the CharacterType.

Reply:
I do not really get why the byte/kind matters here, I only want to get the character length in characters, not bytes, when it is constant at compile time.
Is there already a function for that ?

Reply:
My mistake. I thought this was computing a length in bytes.

A literal CHARACTER string should have a length. A symbol of CHARACTER type will also have its length (either constant or a runtime value). Not sure if that's what you want though.

Reply:
The thing with character expressions is that the length can change in the middle of expressions so we cannot only rely on the symbol map to get the dynamic length information, hence the use of fir.boxchar to propagate everything with dynamic length right now. Agree we should updating the threading of the information in the bridge here.